File: solve/0709_비밀지도/unoesetellar/postforty/[1차] 비밀지도.js
function solution(n, arr1, arr2) {
  var answer = [];

  arr1.forEach((_, i) => {
    const temp = (arr1[i] | arr2[i])
      .toString(2)
      .replaceAll("1", "#")
      .replaceAll("0", " ");

    const result = n > temp.length ? " ".repeat(n - temp.length) + temp : temp;
    answer.push(result);
  });
  return answer;
}

console.log(solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28]));

File: solve/0709_비밀지도/postforty/[1차] 비밀지도.py
# https://school.programmers.co.kr/learn/courses/30/lessons/17681
def solution(n, arr1, arr2):
    answer = []
    for i, _ in enumerate(arr1):
        answer.append(
            f"{bin(arr1[i] | arr2[i])[2:]:0>{n}}".replace("1", "#").replace("0", " ")
        )
    return answer


print(solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28]))
